### PR TITLE
fix: disable selinux labeling

### PR DESCRIPTION
**Pull request by matejvasek**
prevent permission denied on docker socket mount

Signed-off-by: Matej Vasek <mvasek@redhat.com>

### CODE REVIEW

**Comment by matejvasek** [comment]
cc @cardil

File: buildpacks/builder.go [hunk 1]
```diff
@@ -100,3 +124,466 @@ func (builder *Builder) Build(ctx context.Context, f bosonFunc.Function) (err er
...
+// override injecting the security opt
+
+func (c clientWrapper) ContainerCreate(ctx context.Context, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, platform *v1.Platform, containerName string) (container.ContainerCreateCreatedBody, error) {
+	hostConfig.SecurityOpt = []string{"label=disable"}
+	return c.impl.ContainerCreate(ctx, config, hostConfig, networkingConfig, platform, containerName)
+}
+
+// just plain forwards
+
+func (c clientWrapper) ConfigList(ctx context.Context, options types.ConfigListOptions) ([]swarm.Config, error) {
```
Comment (matejvasek) [inline buildpacks/builder.go:141]: Is there any sane way as to do this? Why can't I just override the function I want to. :cry:

**Comment by matejvasek** [comment]
Workaround until https://github.com/buildpacks/pack/issues/1208 is fixed.

**Comment by matejvasek** [comment]
@cardil can you try this out?

**Comment by cardil** [comment]
Works well with Podman 3.1 in rootless mode. I've manage to create, build and deploy a quarkus func on CRC with Serverless 1.15.

**Comment by cardil** [comment]
Fixes #188

**Comment by matejvasek** [comment]
@lkingland @zroubalik could one of  you try this out on macOS?
@jrangelramos  Also it would be good to test this on Windows with actual Windows docker (there is no point to test it against remote docker daemon on linux).

**Comment by matejvasek** [comment]
Before you ask I cannot just `if os == linux` because one can have `DOCKER_HOST` pointing to remote running Linux with SELinux despite running `func` on macOS or Windows

**Comment by matejvasek** [comment]
@rhuss tested this on macOS and it works.

File: buildpacks/builder.go [hunk 2]
```diff
@@ -100,3 +124,466 @@ func (builder *Builder) Build(ctx context.Context, f bosonFunc.Function) (err er
 
 	return
 }
+
+type clientWrapper struct {
+	impl dockerClient.CommonAPIClient
+}
+
```
Comment (zroubalik) [inline buildpacks/builder.go:131]:
```suggestion

// The following section is a workaround until https://github.com/buildpacks/pack/issues/1208 is fixed.
```

**Review by zroubalik — APPROVED** [review]
LGTM, for a reference, I'd add a comment directly to the source code about the workaround.

**Review by lance — APPROVED** [review]

**Pull request merged**